src/qforte/utils/qubit_tapering.py

```diff
@@ -0,0 +1,345 @@
+        for pauli in string.gates():
+            XYZ_gate = pauli.gate_id()
+            trgt = pauli.target()
+            if XYZ_gate in ['Y', 'Z']:
+                prt_chck_mtrx[i, n_qubits - trgt - 1] = 1
+            if XYZ_gate in ['X', 'Y']:
+                prt_chck_mtrx[i, 2 * n_qubits - trgt - 1] = 1
+
+    # 2) Find a basis of the kernel of the parity check matrix
+    # See https://en.wikipedia.org/wiki/Kernel_(linear_algebra)#Computation_by_Gaussian_elimination
+
+    ## Augment the (n_strings) × (2 * n_qubits) parity check matrix by the (2 * n_qubits) × (2 * n_qubits) identity matrix
+
+    prt_chck_mtrx_aug = np.concatenate((prt_chck_mtrx, np.identity(2 * n_qubits, dtype=np.uint8)), axis=0)
+
+    ## Find column echelon form (CEF) of augmented matix.
+
+    ### We only work with the columns of the parity check matrix that have not been transformed yet to CEF.
+    ### This is controlled by the "idx" variable
+    idx = 0
+
+    for row in range(n_strings):
+        ### Find a non-zero row
+        nonzero_indices = np.argwhere(prt_chck_mtrx_aug[row,idx:])
+        if nonzero_indices.shape[0] > 0:
+            column_to_be_CEFed = int(nonzero_indices[0]) + idx
+            ### Bring first non-zero element in this row immediately to the right of that of the previous row
+            prt_chck_mtrx_aug[:, [column_to_be_CEFed , idx]] = prt_chck_mtrx_aug[:, [idx, column_to_be_CEFed]]
+            ### Eliminate remaining non-zero row elements in the part of the parity check matrix that is not in CEF
+            for i in range(1,nonzero_indices.shape[0]):
+                column_not_in_CEF = int(nonzero_indices[i]) + idx
+                prt_chck_mtrx_aug[:,column_not_in_CEF] = np.bitwise_xor(prt_chck_mtrx_aug[:, idx], prt_chck_mtrx_aug[:,column_not_in_CEF])
+            ### decrease the number of columns that need to be checked
+            idx += 1
+
+    ## Find basis vectors of kernel by checking for zero columns in the CEF of the parity check matrix
+
+    dim = False
+    basis = np.zeros((1, 2 * n_qubits), dtype=np.uint8)
+    # The first idx columns are nonzero by construction
+    for column in range(idx, 2 * n_qubits):
+        row = n_strings
+        if not np.any(prt_chck_mtrx_aug[:row, column]):
+            new_basis_vector = prt_chck_mtrx_aug[row:, column].reshape((1,2 * n_qubits))
+            if dim is False:
+                basis = new_basis_vector
+                dim = True
+            else:
+                basis = np.concatenate((basis, new_basis_vector), axis=0)
+
+    # 3) Find maximal Abelian subgroup of the basis of ker{E}
+    # This is done using the symplectic bilinear form B that the binary vector space is equipped with:
+    # B(v,w) = v * B * w.T,
+    # where v and w are binary vectors and B is the matrix of the bilinear form.
+    # B(v,w) = 0 means that v and w commute while B(v,w) = 1 means that they anticommute.
+
+    ## Construct matrix of bilinear form.
```

Comment:
```suggestion
    ## Construct matrix B, which acts on binary vectors. v * B * w.T = 0 means v and w commute.
    ## v * B * w.T = 1 means anticommute.
```

Original comment was verbose.